File: listings/cl-existence-oriented/lastpos.py
from sys import stderr

def print_last_item( items ):
    print( "last item: {}".format( items[ -1 ] ), file = stderr )

def safe_print_last_item( items ):
    if items: print( "last item: {}".format( items[ -1 ] ), file = stderr )

if '__main__' == __name__:
    empty_items = ( )
    items = ( 42, )
    safe_print_last_item( items )
    safe_print_last_item( empty_items )
    print_last_item( items )
    print_last_item( empty_items )  # raises IndexError

File: listings/cl-existence-oriented/lastpos.go
package main

import "fmt"
import "os"

func main() {
	items := []int{42}
	emptyItems := []int{}
	safePrintLastItem(items)
	safePrintLastItem(emptyItems)
	printLastItem(items)
	printLastItem(emptyItems) /* panics */
}

func printLastItem(items []int) {
	fmt.Fprintf(os.Stderr, "last item: %d\n", items[len(items)-1])
}

func safePrintLastItem(items []int) {
	count := len(items)
	if count > 0 {
		lastItem := items[count-1]
		fmt.Fprintf(os.Stderr, "last item: %d\n", lastItem)
	}
}
